src/components/data/data-utils.js

```diff
@@ -949,72 +948,91 @@ export const SingleEntityEditor = ({ entityType, entityName, attributeName, attr
   ])
 }
 
-export const MultipleEntityEditor = ({ entityType, entityNames, attributeNames, entityTypes, workspaceId: { namespace, name }, onDismiss, onSuccess }) => {
+const editEntitiesSetValue = ({ namespace, name }, entityType, entities, attributeToEdit, newValue) => {
+  const entityUpdates = _.map(entity => ({
+    entityType,
+    name: entity.name,
+    operations: [{
+      op: 'AddUpdateAttribute',
+      attributeName: attributeToEdit,
+      addUpdateAttribute: prepareAttributeForUpload(newValue)
+    }]
+  }), entities)
+
+  return Ajax()
+    .Workspaces
+    .workspace(namespace, name)
+    .upsertEntities(entityUpdates)
+}
+
+const editEntitiesConvertType = ({ namespace, name }, entityType, entities, attributeToEdit, newType) => {
+  const { type, entityType: referenceEntityType } = newType
+
+  const entityUpdates = _.map(entity => ({
+    entityType,
+    name: entity.name,
+    operations: [{
+      op: 'AddUpdateAttribute',
+      attributeName: attributeToEdit,
+      addUpdateAttribute: convertAttributeValue(entity.attributes[attributeToEdit], type, referenceEntityType)
+    }]
+  }), entities)
+
+  return Ajax()
+    .Workspaces
+    .workspace(namespace, name)
+    .upsertEntities(entityUpdates)
+}
+
+export const MultipleEntityEditor = ({ entityType, entities, attributeNames, entityTypes, workspaceId: { namespace, name }, onDismiss, onSuccess }) => {
   const [attributeToEdit, setAttributeToEdit] = useState('')
   const [attributeToEditTouched, setAttributeToEditTouched] = useState(false)
   const attributeToEditError = attributeToEditTouched && Utils.cond(
     [!attributeToEdit, () => 'An attribute name is required.'],
     [!_.includes(attributeToEdit, attributeNames), () => 'The selected attribute does not exist.']
   )
 
+  const [operation, setOperation] = useState('setValue')
+
   const [newValue, setNewValue] = useState('')
+  const [newType, setNewType] = useState({ type: 'string' })
```

Comment:
Have you considered enumerating `operation` and `newType` in JSON objects to make valid values more explicit, like so:

```javascript
const newTypes = {
  string: 'string',
  reference: 'reference',
  etc.
}

?

Reply:
There's lots of code in data-utils that uses stringly typed attribute types. I'm open to changing those to use an enum like object, but I think it would make more sense to do in a separate change.

Reply:
Added an enum like object for operation in [12798ce](https://github.com/DataBiosphere/terra-ui/pull/3313/commits/12798ce2db5d1a466b9db425ddd52beef5bd71e2).